aptos-move/framework/aptos-framework/sources/configs/staking_config.move

```diff
@@ -192,6 +338,46 @@ module aptos_framework::staking_config {
         assert!(minimum_stake <= maximum_stake && maximum_stake > 0, error::invalid_argument(EINVALID_STAKE_RANGE));
     }
 
+    fun validate_rewards_config(
+        rewards_rate: u64,
+        min_rewards_rate: u64,
+        rewards_rate_denominator: u64,
+        rewards_rate_period_in_micros: u64,
+        rewards_rate_decrease_rate_bps: u64,
+    ) {
+        assert!(
+            rewards_rate_denominator > 0,
+            error::invalid_argument(EZERO_REWARDS_RATE_DENOMINATOR),
+        );
+
+        // `rewards_rate` and `min_rewards_rate` which are the numerator are limited to
+        // be `<= MAX_REWARDS_RATE` in order to avoid the arithmetic overflow in the rewards calculation.
+        // `rewards_rate_denominator` can be adjusted to get the desired rewards rate
+        // (i.e., rewards_rate / rewards_rate_denominator).
+        assert!(rewards_rate <= MAX_REWARDS_RATE, error::invalid_argument(EINVALID_REWARDS_RATE));
+        assert!(min_rewards_rate <= MAX_REWARDS_RATE, error::invalid_argument(EINVALID_REWARDS_RATE));
+
+        assert!(
+            min_rewards_rate <= rewards_rate,
+            error::invalid_argument(EINVALID_MIN_REWARDS_RATE)
+        );
+
+        // We assert that (rewards_rate / rewards_rate_denominator <= 1).
+        assert!(rewards_rate <= rewards_rate_denominator, error::invalid_argument(EINVALID_REWARDS_RATE));
+
+        // Rewards rate decrease rate cannot be greater than 100%. Otherwise rewards rate will be negative.
+        assert!(
+            rewards_rate_decrease_rate_bps <= BPS_DENOMINATOR,
+            error::invalid_argument(EINVALID_REWARDS_RATE_DECREASE_RATE)
+        );
+
+        // Now rewards rate decrease interval must be 1 year.
+        assert!(
+            rewards_rate_period_in_micros == ONE_YEAR_IN_MICROS,
```

Comment:
I want to keep `rewards_rate_period_in_micros` on chain because we cannot modify a struct. In case we want to change the parameter, we don't need to add a new struct.

However the logic to handle rewards_rate_period_in_micros change might be tricky. For example, since last time, it has already passed several **new** rewards rate periods at the time we change it. Considering that it's out of scope of our current priority, I made `rewards_rate_period_in_micros` immutable at the moment.

Reply:
1. I think `rewards_rate_period` should >= `epoch_interval`. But `epoch_interval` is in `block.move` so reading this value in `staking_config.move` might cause a cyclic dependency. Also enforcing this requires changes in both `block.move` and `staking_config.move`.
2. "old rate has passed through several periods" might bring some undefined behaviors. For example, should we reset `last_rewards_rate_period_start_in_micros` or decrease rewards based on the number of passed rewards period? I feel it's more like a product question which we can figure out when we really need it. For now I just want to have a placeholder field to avoid potential unnecessary struct adding.

Reply:
Sure that's reasonable. Can you leave a comment here documenting this and that this can change in the future to allow setting a period that's not 1 year?